CRM/Dedupe/Merger.php

```diff
@@ -1812,10 +1812,9 @@ public static function moveAllBelongings($mainId, $otherId, $migrationInfo) {
         $query = "UPDATE civicrm_contact SET external_identifier = null WHERE id = {$otherId}";
         CRM_Core_DAO::executeQuery($query);
       }
-
       civicrm_api3('contact', 'delete', array('id' => $otherId));
-      CRM_Core_BAO_PrevNextCache::deleteItem($otherId);
     }
+    CRM_Core_BAO_PrevNextCache::deleteItem($otherId);
```

Comment:
I'm feeling pretty uncomfortable with the permission check in this function as it seems that if they don't have permission to merge that should be checked earlier on. Regardless - if the belongings have been shifted we should pull the semi-deleted contact out of any sets it belongs to. If we leave it in the merge set we can create a code loop in the unit tests & we can reasonably assume it's fairly invalid in most searches now. However, I would argue we should never reach this point in that only contacts who can run a test should be able to get to the move all belongings fn